packages/xdl/src/Project.ts

```diff
@@ -661,6 +727,38 @@ export async function exportForAppHosting(
     );
   }
 
+  if (
+    validPostExportHooks.length ||
+    (exp.ios && exp.ios.publishManifestPath) ||
+    (exp.android && exp.android.publishManifestPath) ||
+    EmbeddedAssets.shouldEmbedAssetsForExpoUpdates(projectRoot, exp, pkg, target)
+  ) {
```

Comment:
It seems like these `if` conditions are unnecessary – we could just always do `for (const hook of validPostExportHooks) {` since it will be a no-op if `validPostExportHooks` is empty.

```suggestion
```